dvc/repo/index.py

```diff
@@ -128,10 +128,14 @@ def __init__(
         self._params = params or {}
         self._collected_targets: Dict[int, List["StageInfo"]] = {}
 
-    def __repr__(self) -> str:
-        rev = "workspace"
+    @cached_property
+    def rev(self) -> Optional[str]:
         if not isinstance(self.repo.fs, LocalFileSystem):
-            rev = self.repo.get_rev()[:7]
+            return self.repo.get_rev()[:7]
```

Comment:
just to make sure: it'll be always a hash right? (no matter if there are tags for example)

Reply:
asking since if we want to use it for cache it should be stable ....

Reply:
yep, always hash from what I can tell. Maybe would use a long rev for actual caching, but this does it for now to start playing with it.